tests/integration/suite/placement/dissemination/namespaced_actors.go

```diff
@@ -0,0 +1,313 @@
+	"github.com/dapr/dapr/tests/integration/suite"
+)
+
+func init() {
+	suite.Register(new(namespacedActors))
+}
+
+type namespacedActors struct {
+	place *placement.Placement
+}
+
+func (n *namespacedActors) Setup(t *testing.T) []framework.Option {
+	n.place = placement.New(t)
+
+	return []framework.Option{
+		framework.WithProcesses(n.place),
+	}
+}
+
+func (n *namespacedActors) Run(t *testing.T, ctx context.Context) {
+	n.place.WaitUntilRunning(t, ctx)
+
+	t.Run("actors in different namespaces are disseminated properly", func(t *testing.T) {
+		host1 := &v1pb.Host{
+			Name:      "myapp1",
+			Namespace: "ns1",
+			Port:      1231,
+			Entities:  []string{"actor1", "actor10"},
+			Id:        "myapp1",
+			ApiLevel:  uint32(20),
+		}
+		host2 := &v1pb.Host{
+			Name:      "myapp2",
+			Namespace: "ns2",
+			Port:      1232,
+			Entities:  []string{"actor2", "actor3"},
+			Id:        "myapp2",
+			ApiLevel:  uint32(20),
+		}
+		host3 := &v1pb.Host{
+			Name:      "myapp3",
+			Namespace: "ns2",
+			Port:      1233,
+			Entities:  []string{"actor4", "actor5", "actor6", "actor10"},
+			Id:        "myapp3",
+			ApiLevel:  uint32(20),
+		}
+
+		ctx3, cancel3 := context.WithCancel(ctx)
+		placementMessageCh1 := n.place.RegisterHost(t, ctx, host1)
+		placementMessageCh2 := n.place.RegisterHost(t, ctx, host2)
+		placementMessageCh3 := n.place.RegisterHost(t, ctx3, host3)
+
+		// Host 1
+		msgNumber := 0
+		require.EventuallyWithT(t, func(t *assert.CollectT) {
+			select {
+			case <-ctx.Done():
+				return
+			case placementTables := <-placementMessageCh1:
+				if ctx.Err() != nil {
+					return
+				}
+				msgNumber++
+				if msgNumber == 1 {
+					require.Empty(t, placementTables.GetEntries())
+				}
+				if msgNumber == 2 {
+					require.Len(t, placementTables.GetEntries(), 2)
+					require.Contains(t, placementTables.GetEntries(), "actor1")
+					require.Contains(t, placementTables.GetEntries(), "actor10")
+
+					loadMap := placementTables.GetEntries()["actor10"].GetLoadMap()
+					require.Len(t, loadMap, 1)
+					require.Contains(t, loadMap, host1.GetName())
+				}
+			}
+
+			assert.Equal(t, 2, msgNumber)
+		}, 10*time.Second, 10*time.Millisecond)
+
+		// Host 2
+		msgNumber = 0
+		require.EventuallyWithT(t, func(t *assert.CollectT) {
+			select {
+			case <-ctx.Done():
+				return
+			case placementTables := <-placementMessageCh2:
+				if ctx.Err() != nil {
+					return
+				}
+				msgNumber++
+				if msgNumber == 1 {
+					assert.Empty(t, placementTables.GetEntries())
+				}
+				if msgNumber == 2 {
+					require.Len(t, placementTables.GetEntries(), 6)
+					require.Contains(t, placementTables.GetEntries(), "actor2")
+					require.Contains(t, placementTables.GetEntries(), "actor3")
+					require.Contains(t, placementTables.GetEntries(), "actor4")
+					require.Contains(t, placementTables.GetEntries(), "actor5")
+					require.Contains(t, placementTables.GetEntries(), "actor6")
+					require.Contains(t, placementTables.GetEntries(), "actor10")
```

Comment:
I think `require` will cause a panic when an assertion fails inside a `EventuallyWithT` loop. Consider using `assert`, with an `if` check on the assertion to gate further assertions to avoid a nil pointer panic .

https://github.com/dapr/dapr/blob/58edd57c5c16dcbccbfb3673b2b2c2e20b300ef5/tests/integration/suite/actors/reminders/rebalancing.go#L123

Reply:
The requires should be ok, because they're under an `if` condition, but in the eventual case of them failing in the future, it's better to have them as asserts.
I also added a check for the element's presence.